godot-core/src/init/mod.rs

```diff
@@ -244,6 +244,38 @@ fn gdext_on_level_deinit(level: InitLevel) {
 /// Note that this only changes the name. You cannot provide your own function -- use the [`on_level_init()`][ExtensionLibrary::on_level_init]
 /// hook for custom startup logic.
 ///
+/// # Custom Wasm binary filename
+/// Upon exporting a game to the web, the library has to know at runtime the exact name of the `.wasm` binary file being used to load
+/// each GDExtension. By default, Rust exports the binary as `cratename.wasm`, so that is the name checked by godot-rust by default.
+///
+/// However, if you need to rename that binary, you can make the library aware of the new binary name with
+/// `#[gdextension(wasm_binary = "newname.wasm")]` (don't forget to include the `.wasm` extension).
+///
+/// For example, to have two simultaneous versions, one supporting multi-threading and the other not, you could add a suffix to the
+/// filename of the Wasm binary of the multi-threaded version in your build process. If you choose the suffix `.threads.wasm`, you're
+/// in luck as godot-rust already accepts this suffix by default. But let's say you want to use a different suffix, such as
+/// `-with-threads.wasm`. For this, you can have a `"nothreads"` feature which, when absent, should produce a suffixed binary, which
+/// can be informed to gdext as follows:
+///
+/// ```no_run
+/// # use godot::init::*;
+/// struct MyExtension;
+///
+/// // Tell gdext we add a custom suffix to the binary with thread support
+/// // Please note that this is not needed if "mycrate.threads.wasm" is used
+/// #[cfg(not(feature = "nothreads"))]
+/// #[gdextension(wasm_binary = "mycrate-with-threads.wasm")]
+/// unsafe impl ExtensionLibrary for MyExtension {}
+///
+/// // Binary name unchanged (mycrate.wasm) without thread support
+/// #[cfg(feature = "nothreads")]
+/// #[gdextension]
+/// unsafe impl ExtensionLibrary for MyExtension {}
+/// ```
```

Comment:
I think the example is great and also readable, maybe mention briefly in a comment or the prose text, that `cfg_attr` could also be used as an alternative.

Reply:
(Not applicable anymore after the latest changes.)